addons/product/models/product_catalog_mixin.py

```diff
@@ -91,7 +91,7 @@ def _get_product_catalog_order_line_info(self, product_ids, child_field=False, *
         order_line_info = {}
         default_data = self._default_order_line_values(child_field)
 
-        for product, record_lines in self._get_product_catalog_record_lines(product_ids).items():
+        for product, record_lines in self._get_product_catalog_record_lines(product_ids, child_field).items():
```

Comment:
```suggestion
        for product, record_lines in self._get_product_catalog_record_lines(product_ids, chield_field=child_field).items():
```

?
That way, you don't have to modify the overrides in repair and sale ?

Reply:
Hello @Feyensv thanks for suggestion,I have updated the code accordingly.